pandas/tests/tslibs/test_parsing.py

```diff
@@ -226,3 +226,29 @@ def test_parse_time_string_check_instance_type_raise_exception():
     result = parse_time_string("2019")
     expected = (datetime(2019, 1, 1), "year")
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "fmt,expected",
+    [
+        ("%Y %m %d %H:%M:%S", True),
+        ("%Y/%m/%d %H:%M:%S", True),
+        (r"%Y\%m\%d %H:%M:%S", True),
+        ("%Y-%m-%d %H:%M:%S", True),
+        ("%Y.%m.%d %H:%M:%S", True),
+        ("%Y%m%d %H:%M:%S", True),
+        ("%Y-%m-%dT%H:%M:%S", True),
+        ("%Y-%m-%dT%H:%M:%SZ", True),
```

Comment:
This doesn't appear like a valid strftime directive. To parse `Z`, I think `%z` needs to be used e.g.

```
In [9]: datetime.datetime.strptime("2020-05-01T18:04:03Z", "%Y-%m-%dT%H:%M:%S%z")
Out[9]: datetime.datetime(2020, 5, 1, 18, 4, 3, tzinfo=datetime.timezone.utc)
```

Reply:
Thank you for your comment.
I fixed it to fit the output of `guess_datetime_format`.

ex.
```
In [1]: from pandas._libs.tslibs.parsing import guess_datetime_format
   ...: guess_datetime_format('2020-05-01T18:04:03Z')
Out[1]: '%Y-%m-%dT%H:%M:%SZ'
````

But as you said, the format string like "%SZ" should be "%S%z"
So now I guess the `guess_datetime_format` function also has some bugs.